app/controllers/rankings_controller.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class RankingsController < ApplicationController
+  before_action :check_closing_time, only: [:index]
+  def index
+    @category = params[:category].presence || 'food'
+
+    @rankings = Vote.ranking_by_single(@category)
+    @rankings_total = Vote.ranking_by_total_votes(@category)
+    @comments = Vote.fetch_comment(@category)
+  end
+
+  private
+
+  def check_closing_time
+    return unless Time.current < CLOSING_TIME
+
+    redirect_to root_path, alert: 'このページは現在公開されていません。'
```

Comment:
ここは変にguard clause使わずにifで書いた方が読みやすい気がします


```suggestion
    if Time.current >= CLOSING_TIME
      redirect_to root_path, alert: 'このページは現在公開されていません。'
    end
```


あるいは、voteクラスのクラスメソッドに`closed?`を作成して`Vote.closed?`といった形で使いまわしても良いかも

イメージ
```
class Vote
  CLOSING_TIME = Time.zone.parse('2024-09-29 19:00')

  def self.closed?
    Time.current >= CLOSING_TIME
  end
end
```

Reply:
```
   if Vote.open?
      redirect_to root_path, alert: 'このページは現在公開されていません。'
    end
```
ifのtrueの時の実行結果が投票可能のときの内容のため、open?メソッドにしました。

Reply:
ありがとうございます。`Vote.closed?`として書き換えました。またunlessではなくifを使いました。